pkg/collector/python/init.go

```diff
@@ -298,20 +298,26 @@ func resolvePythonExecPath(pythonVersion string) string {
 	// For Windows, the binary should be in our path already and have a
 	// consistent name
 	if runtime.GOOS == "windows" {
+		if PythonHome != "" {
+			return filepath.Join(PythonHome, PythonWinExeBasename), nil
+		}
+
 		// If we are in a development environment, PythonHome will not be set so we
 		// use the absolute path to the python.exe in our path.
-		if PythonHome == "" {
-			log.Warnf("Python home is empty. Inferring interpreter path from binary in path.")
-			interpreterAbsPath, err := resolveExecPath(PythonWinExeBasename)
-			if err != nil {
+		log.Warnf("Python home is empty. Inferring interpreter path from binary in path.")
+		interpreterAbsPath, err := resolveExecPath(PythonWinExeBasename)
+		if err != nil {
+			if allowFailure {
 				log.Warnf("Error trying to resolve interpreter path for executable: '%v'."+
 					" Python's 'multiprocessing' library may fail to work.", err)
+				return PythonWinExeBasename, nil
+			} else {
+				return "", fmt.Errorf("Error trying to resolve interpreter path for executable: '%v'."+
```

Comment:
I think currently '%v' going to be the `err` value not the executable name, so this error might look a little weird.

Also, since we're using pretty much the same error message in two places and for consistency and maintainability, we could maybe implement a custom error. Feel free to ignore, just a nit.

Reply:
[Fixed](https://github.com/DataDog/datadog-agent/pull/8369/commits/e10fde45b48576c9e71f8dfc8aa0decba744f82a)